### PR TITLE
New: Adding Open With skeleton, wrapper, and test file

### PR DESCRIPTION
**Pull request by jeremypress**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jeremypress has signed the CLA. Thanks for the pull request!

**Comment by boxcla** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Percentage Diff | Before | After |
| ------------- |:----:|:---------------:|:------:|:-----:|
| Total | 0 B | 0.00% | 0 B | 0 B |

Compared against previously merged PR #509

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**

File: src/components/OpenWith/OpenWith.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+     * [constructor]
+     *
+     * @private
+     * @return {OpenWith}
+     */
+    constructor(props: Props) {
+        super(props);
+        const { token, apiHost, clientName, requestInterceptor, responseInterceptor } = props;
+
+        this.id = uniqueid('bcow_');
```
Comment (priyajeet) [inline src/components/OpenWith/OpenWith.js]: bcow? lol

File: src/components/OpenWith/OpenWith.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+     * @inheritdoc
+     * @return {Element}
+     */
+    render() {
+        const { language, messages: intlMessages }: Props = this.props;
+
+        // Placeholder
+        return (
+            <Internationalize language={language} messages={intlMessages}>
+                <button> Open With </button>
```
Comment (priyajeet) [inline src/components/OpenWith/OpenWith.js]: Ship it!

File: src/components/OpenWith/OpenWith.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+import React, { PureComponent } from 'react';
+import uniqueid from 'lodash/uniqueId';
+import API from '../../api';
+import Internationalize from '../Internationalize';
+
+import { DEFAULT_HOSTNAME_API, CLIENT_NAME_OPEN_WITH } from '../../constants';
+
+type Props = {
+    fileId?: string,
```
Comment (priyajeet) [inline src/components/OpenWith/OpenWith.js]: Probably comment it, so that styleguide gets it

**Pull request merged**